### PR TITLE
[4556] [studio] Implement Available Actions per the new design

### PR DESCRIPTION
**Pull request by dejan-brkic**
Added protected folders configuration

### Ticket reference or full description of what's in the PR
https://github.com/craftercms/craftercms/issues/4556

### CODE REVIEW

File: src/main/java/org/craftercms/studio/impl/v2/security/SemanticsAvailableActionsResolverImpl.java [hunk 1]
```diff
@@ -84,6 +92,20 @@ private long applySpecialUseCaseFilters(String username, String siteId, Item ite
             throws ServiceLayerException, UserNotFoundException {
         long result = availableActions;
 
+        if (StringUtils.equals(item.getPath(), HOME_PAGE_PATH)) {
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/studio/impl/v2/security/SemanticsAvailableActionsResolverImpl.java:95]: Why isn't this a pattern like the others?

**Review by sumerjabri — CHANGES_REQUESTED** [review]

**Review by sumerjabri — CHANGES_REQUESTED** [review]
This PR seems to contain other stuff not related to the ticket.

File: src/main/java/org/craftercms/studio/impl/v2/security/SemanticsAvailableActionsResolverImpl.java (hunk 1 above)
Comment (dejan-brkic) [reply src/main/java/org/craftercms/studio/impl/v2/security/SemanticsAvailableActionsResolverImpl.java:95]: hardcoded home page ... it is always on the same path and does not need to be configured. If configured, somebody can remove protection by accident or intentionally

**Comment by dejan-brkic** [comment]
> This PR seems to contain other stuff not related to the ticket.

Merge conflicts and Codacy fixes

**Review by sumerjabri — APPROVED** [review]

**Pull request merged**